Automattic/wp-calypso
# DesignPickerStep uses feature check for premium themes

#### Changes proposed in this Pull Request

* DesignPickerStep uses feature check for premium themes
  * ~~Waiting on #63673 since the step has an inaccurate view of your site's features if you purchased pro~~ Done!

#### Testing instructions

* Use this PR
* Visit `/start?flags=-signup/stepper-flow` to create a new site with stepper turned off
* Enter a domain name, click the "Select" button for the wp.com subdomain
* Click "start with a free site" near the top
* See intent screen (write/build/sell): choose build
* You should now be on `/start/setup-site/design-setup-site`
  * If you are on `/setup/designSetup` instead, that is the stepper version which is not covered by PR. Turning off stepper didn't work correctly. Try again or try something more drastic, like the inverse of a599ca3f locally.
* Scroll through the list of themes and find a premium theme.
* The top button should read "Upgrade Plan" and the tooltip on the star should not say it is included in your plan.

![2022-05-17_13-12](https://user-images.githubusercontent.com/937354/168887465-20e43c85-7a95-4ffa-b847-1e1a848655ff.png)

* Repeat the steps for a new site, but this time, pick the Pro plan after choosing a domain, and checkout.
* This time, when seeing a premium theme, the button should read "Start with" and the tooltip on the star says it's included in your plan.

![2022-05-17_13-17](https://user-images.githubusercontent.com/937354/168887601-035aa779-97d8-4a22-a085-ff66caab8d87.png)



Related to p4TIVU-a66-p2
